runtime/lua/vim/lsp/buf.lua

```diff
@@ -726,6 +742,14 @@ end
 ---           If in visual mode this defaults to the active selection.
 ---           Table must contain `start` and `end` keys with {row,col} tuples
 ---           using mark-like indexing. See |api-indexing|
+---  - sync:  (boolean|nil)
```

Comment:
nit: The `vim.lsp.buf.format(...)` param is called `async`. It would be nice to keep the two consistent.

Reply:
You're right, but I'm leaning towards implementing this as a separate method instead, because:

 * the defaults are different - `.code_action()` is async by default, but `.format()` is sync by default. I think it would still feel inconsistent.
 * if it's a separate method, it can return a `result, err`.
 
... but it sounds like this is likely not going to be accepted either way. If I (or someone) doesn't think of a way to do this which satisfies the direction described by @mfussenegger , I'm probably going to close this PR.